libdnf/rpm/package_sack.cpp

```diff
@@ -187,6 +188,14 @@ void PackageSack::Impl::setup_excludes_includes(bool only_main) {
     }
 }
 
+void PackageSack::Impl::add_excludes(const PackageSet & excludes) {
```

Comment:
When this method is added then we have to ensure that `add_excludes()` and `setup_excludes_includes()` can be used in any order.

Reply:
Good catch - currently `setup_excludes_includes` resets excludes only in case that there are some defined in configuration. In case there is no exclude in configuration, the method doesn't do anything.

Reply:
I've changed `setup_excludes_includes` to always reset excludes.

Reply:
Be honest, I am not sure whether reset is a good idea. In case of reset we have to force API user to call methods in certain order.

Reply:
Reset in the sense of set them according the config files - which it didn't in case the config files did not contain any `excludepkgs=...` field.
Well, of course they need to call it in certain order - first setup_excludes_includes to (re)set excludes according the configuration and then use custom additions / removals.
It's true that name `setup_excludes_includes` is kind of misleading. Something like `read_excludes_includes_from_config` would be more descriptive, but also awfully long :)

Reply:
The other solution may be to adjust `setup_excludes_includes` to add excludes from configuration instead of setting it. Than the call order would not matter.

Reply:
I've added a commit that renames setup_excludes_includes to add_excludes_includes_from_config and changes behaviour accordingly. Now the methods can be called in any order.